### PR TITLE
Use resource framework to free up open dirs

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: src/primitive_file_posix.cc [hunk 1]
```diff
@@ -145,35 +145,36 @@ PRIMITIVE(open) {
   return Smi::from(fd);
 }
 
-class Directory {
+class Directory : public SimpleResource {
  public:
   TAG(Directory);
-  DIR* dir;
+  Directory(SimpleResourceGroup* group, DIR* dir) : SimpleResource(group), _dir(dir) { }
+  ~Directory() { closedir(_dir); }
```
Comment (kasperl) [inline src/primitive_file_posix.cc:152]: What happens if _dir is null?

File: packages/host/src/directory.toit [hunk 2]
```diff
@@ -95,17 +95,17 @@ class DirectoryStream:
...
+      byte_array := readdir_ dir_
       if not byte_array: return null
       str := byte_array.to_string
       if str == "." or str == "..": continue
       return str
 
   close -> none:
-    fd := fd_
-    closedir_ fd
+    dir := dir_
```
Comment (kasperl) [inline packages/host/src/directory.toit:105]: What's up with this dance? Should we set dir_ to null here?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/primitive_file_posix.cc (hunk 1 above)
Comment (erikcorry) [reply src/primitive_file_posix.cc:152]: That can't happen.

File: packages/host/src/directory.toit (hunk 2 above)
Comment (erikcorry) [reply packages/host/src/directory.toit:105]: That depends on whether we want to silently allow multiple close calls.  I think in general that's just nicer.  If we leave it as is the underlying Directory is deleted, but the proxy lives on with a null pointer in it.  That means we get an ALREADY_CLOSED exception if we call close twice.

**Pull request merged**